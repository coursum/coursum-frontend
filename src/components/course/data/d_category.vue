<template>
  <div
    class="px-6 text-truncate"
  >
    {{ categoryData ? categoryData : $t("others") }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { basicTemplate } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DCategory',
  props: {
    category: { type: Object, default: basicTemplate },
  },
  computed: {
    categoryData(): string | null | undefined {
      return this.category?.[this.curLang];
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
div{
  font-size: 0.7rem;
  color: gray;
}
</style>

<i18n>
{
  "en": {
    "others": "others"
  },
  "jp": {
    "others": "その他"
  }
}
</i18n>
