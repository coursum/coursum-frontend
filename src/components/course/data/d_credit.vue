<template>
  <v-chip
    v-if="credit"
    style="color: #929292"
    outlined
    x-small
  >
    {{ credit }}{{ $t("credit") }}
  </v-chip>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DCredit',
  props: {
    credit: {
      type: Number,
      default: undefined,
    },
  },
});
</script>

<i18n>
{
  "en": {
    "credit": "credit"
  },
  "jp": {
    "credit": "単位"
  }
}
</i18n>
