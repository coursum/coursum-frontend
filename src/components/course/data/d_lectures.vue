<template>
  <div
    v-if="lecturers && 0 < lecturers.length"
    :class="{'text-truncate': textTruncate}"
    class="px-6"
  >
    <template
      v-for="(lecturer, i) in lecturers"
    >
      <span
        v-if="lecturer.name && lecturer.name[`${curLang}`]"
        :key="i"
        style="display:inline-block;"
        class="mr-2"
      >
        {{ lecturer.name[`${curLang}`] }}
      </span>
    </template>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DLectures',
  props: {
    lecturers: {
      type: Array,
      default() {
        return undefined;
      },
    },
    textTruncate: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
span{
  font-size: 0.85rem;
  color: var(--v-wordcolor-base);
}
</style>
