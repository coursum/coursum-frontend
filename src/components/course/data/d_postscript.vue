<template>
  <span
    v-if="postscriptData"
    style="color: #929292"
    class="px-6 text-caption text-truncate"
  >
    {{ postscriptData }}
  </span>
</template>

<script lang="ts">
import { PropType, computed, defineComponent } from '@vue/composition-api';

import { Basic, basicTemplate } from '@/assets/CourseInfo';

export default defineComponent({
  name: 'DPostsctipt',
  props: {
    postscript: {
      type: Object as PropType<Basic>,
      default: basicTemplate,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'jp' | 'en');

    const postscriptData = computed((): string | null | undefined => props
      .postscript?.[curLang.value]);

    return {
      postscriptData,
    };
  },
});
</script>
