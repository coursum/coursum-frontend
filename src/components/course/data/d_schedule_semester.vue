<template>
  <v-chip v-if="semesterData"
          color="caption"
          outlined
          x-small
  >
    {{ semesterData }}
  </v-chip>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';
import type { Course } from 'coursum-types';

export default defineComponent({
  name: 'DScheduleSemester',
  props: {
    semester: {
      type: Object as PropType<Course['schedule']['semester']>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'ja');
    const semesterData = computed(() => props.semester[curLang.value]);

    return {
      semesterData,
    };
  },
});
</script>
