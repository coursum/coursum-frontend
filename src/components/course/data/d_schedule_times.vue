<template>
  <span v-if="courseTimes">
    <v-chip v-for="(time, i) in courseTimes" :key="i"
            color="caption"
            outlined
            x-small
    >
      {{ time }}
    </v-chip>
  </span>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';
import type { Course } from 'coursum-types';

export default defineComponent({
  name: 'DScheduleTimes',
  props: {
    times: {
      type: Object as PropType<Course['schedule']['times']>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'ja');

    const courseTimes = computed(() => {
      const times = props.times[curLang.value] || '';

      return times.split(',').map((time) => time.trim());
    });

    return {
      courseTimes,
    };
  },
});
</script>
