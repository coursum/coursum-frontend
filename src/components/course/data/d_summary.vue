<template>
  <div
    v-if="summaryData"
    :class="{summary: textTruncate}"
    class="px-6 primary--text py-1"
    :style="summaryStyle"
  >
    <p
      class="my-0"
    >
      {{ summaryData }}
    </p>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { basicTemplate } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DSummary',
  props: {
    summary: {
      type: Object,
      default: basicTemplate,
    },
    title: {
      type: Object,
      default: basicTemplate,
    },
    textTruncate: {
      type: Boolean,
      default: true,
    },
  },
  data() {
    return {
      summaryStyle: { 'font-size': '1rem' },
    };
  },
  computed: {
    summaryData(): string | null | undefined {
      if (this.titleJp === '研究会Ａ' || this.titleJp === '研究会Ｂ') {
        return this.summary?.en;
      }

      return this.summary?.[this.curLang];
    },
    titleJp(): string | null | undefined {
      return this.title?.jp;
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
@import url('https://fonts.googleapis.com/css2?family=Noto+Sans+JP&display=swap');

.summary p {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 5;
  overflow: hidden;
}
</style>
