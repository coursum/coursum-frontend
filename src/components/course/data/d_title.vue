<template>
  <div
    v-if="titleData"
    class="text-truncate font-weight-bold word--text"
  >
    {{ titleData }}
  </div>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';

import type { Basic } from '@/types/CourseInfo';

export default defineComponent({
  name: 'DTitle',
  props: {
    title: {
      type: Object as PropType<Basic>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'jp');

    const titleData = computed((): string | null | undefined => props.title?.[curLang.value]);

    return {
      titleData,
    };
  },
});
</script>

<style scoped>
div {
  font-size: 0.9rem;
}
</style>
