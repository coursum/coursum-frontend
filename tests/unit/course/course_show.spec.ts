import CourseShow from '@/components/course/course_show.vue';
import { shallowMount, mount } from '@vue/test-utils';
import { i18n } from '../../setup';

import { courseDataMock } from '../../data';

const options = {
  i18n,
  propsData: {
    courseData: courseDataMock,
  },
  mocks: {
    $t: (msg: string) => msg,
  },
};

describe('CourseShow', () => {
  it('should render correct markup when prop is correct', () => {
    const wrapper = shallowMount(CourseShow, options);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correct markup when prop is correct', () => {
    const wrapper = mount(CourseShow, options);

    expect(wrapper.element).toMatchSnapshot();
  });
});
