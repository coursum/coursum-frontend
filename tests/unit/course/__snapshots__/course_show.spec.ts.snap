// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseShow should render correct markup when prop is correct 1`] = `
<v-hover-stub
  closedelay="0"
  opendelay="0"
>
  <v-card-stub
    class="my-2 rounded-lg d-flex flex-column justify-space-between elevation-1 transition-swing"
    loaderheight="4"
    tag="div"
    to="/course/name jp name jp"
  >
    <div
      class="rounded-t-lg py-2 title-bg"
    >
      <d-category-stub
        category="[object Object]"
      />
       
      <d-lectures-stub
        lecturers="[object Object]"
        texttruncate="true"
      />
       
      <d-title-stub
        title="[object Object]"
      />
    </div>
     
    <d-postscript-stub
      postscript="[object Object]"
    />
     
    <d-summary-stub
      class="py-2"
      summary="[object Object]"
      texttruncate="true"
      title="[object Object]"
    />
     
    <div
      class="px-4 py-2"
    >
      <d-credit-stub
        credit="2"
      />
       
      <d-schedule-semester-stub
        semester="[object Object]"
      />
       
      <d-schedule-times-stub
        times="[object Object]"
      />
    </div>
  </v-card-stub>
</v-hover-stub>
`;

exports[`CourseShow should render correct markup when prop is correct 2`] = `
<router-link
  class="my-2 rounded-lg d-flex flex-column justify-space-between v-card v-card--link v-sheet theme--light elevation-1 transition-swing"
  tabindex="0"
>
  <div
    class="rounded-t-lg py-2 title-bg"
  >
    <div
      class="px-6 text-truncate"
      style="font-size: 0.7rem; color: white;"
    >
      
  category en

    </div>
     
    <div
      class="px-6 text-truncate"
    >
      <span
        class="mr-2"
        style="display: inline-block; font-size: 0.98rem; color: white;"
      >
        
      name en
    
      </span>
    </div>
     
    <div
      class="px-6 font-weight-black text-truncate"
      style="font-size: 1.1rem; color: white;"
    >
      
  name en

    </div>
  </div>
   
  <span
    class="px-6 text-caption text-truncate"
    style="color: rgb(146, 146, 146);"
  >
    
  postscript en

  </span>
   
  <div
    class="px-6 primary--text py-1 py-2 summary"
    style="font-size: 1rem;"
  >
    <p
      class="my-0"
    >
      
    summary en
  
    </p>
  </div>
   
  <div
    class="px-4 py-2"
  >
    <span
      class="v-chip v-chip--no-color v-chip--outlined theme--light v-size--x-small"
      style="color: rgb(146, 146, 146);"
    >
      <span
        class="v-chip__content"
      >
        
  2credit

      </span>
    </span>
     
    <span
      class="v-chip v-chip--no-color v-chip--outlined theme--light v-size--x-small"
      style="color: rgb(146, 146, 146);"
    >
      <span
        class="v-chip__content"
      >
        
  semester en

      </span>
    </span>
     
    <span>
      <span
        class="v-chip v-chip--outlined theme--light v-size--x-small primary primary--text"
      >
        <span
          class="v-chip__content"
        >
          
    æœˆ1
  
        </span>
      </span>
    </span>
  </div>
</router-link>
`;
